ownCloudAppShared/AppLock/AppLockManager.swift

```diff
@@ -153,11 +159,13 @@ public class AppLockManager: NSObject {
 		userDefaults = OCAppIdentity.shared.userDefaults!
 
 		super.init()
+		significantTimeChangeOccurred()
```

Comment:
By handling every initialization of the `AppLockManager` as a significant time change and resetting the countdown, users may be hit without trying to bypass the lock timer, f.ex. during regular usage like:
- invoking the share extension from a share sheet
- iOS quits the app in the background to free memory for apps launched after it 
- the user closes the app

Adding `UIApplication.significantTimeChangeNotification` as a signal to the mix of detecting a bypass attempt is a good idea, but if the notification is not delivered to the app and/or app extensions when they aren't running at the time the significant time change occurs, it can't be relied on alone.

Resetting the timer every time the `AppLockManager` is initialized in a new process closes the gaps that a `UIApplication.significantTimeChangeNotification` can leave, but also has negative effects on legitimate usage, unfortunately, which I think we should avoid.

Why not use `ProcessInfo.processInfo.systemUptime` as outlined above?

Reply:
@felix-schwarz please go ahead if by implementing the prevention using `ProcessInfo.processInfo.systemUptime` if you think it is a better solution.

Reply:
Ok, I implemented a solution based on `ProcessInfo.processInfo.systemUptime` at https://github.com/owncloud/ios-app/pull/1347 . If it passes testing on a real device, I'd suggest to close this PR and proceed with #1347 instead.